backends/ebpf/ebpftargets.py

```diff
@@ -257,70 +284,70 @@ def _load_ebpf(self, ebpfdir, ifname):
         args = ["tc"]
         # Launch tc to load the ebpf object to the specified interface
         args.extend(["filter", "add", "dev", ifname, "ingress",
-                     "bpf", "da", "obj", self.template + ".o",
-                     "sec", "prog", "verb"])
+                     "bpf", "da", "obj", self.template + "_ebpf.o",
+                     "section", "prog", "verbose"])
         # The bpf program to attach to the interface
         errmsg = "Failed to load the eBPF byte code using tc:"
         return run_timeout(self.options, args, TIMEOUT, self.outputs, errmsg)
 
+    def _create_runtime(self):
+        args = self.get_make_args(self.ebpfdir, self.options.target)
+        # List of bpf programs to attach to the interface
+        args.append("BPFOBJ=" + self.template)
+        args.append("CFLAGS+=-DCONTROL_PLANE")
+        args.append("SOURCES=")
+        errmsg = "Failed to build the filter:"
+        return run_timeout(self.options, args, TIMEOUT, self.outputs, errmsg)
+
     def create_filter(self):
         require_root(self.outputs)
         # Create interface with unique id (allows parallel tests)
         # TODO: Create one custom interface per pcap file we parse
         ifname = str(os.getpid())
         self._create_bridge(ifname)
         # Use clang to compile the generated C code to a LLVM IR
-        result = self._compile_ebpf(self.ebpfdir)
+        result = self._compile_ebpf_object()
         if result != SUCCESS:
             self._remove_bridge(ifname)
             return result
-        result = self._load_ebpf(self.ebpfdir, ifname)
+        result = self._load_ebpf(ifname)
+        if result != SUCCESS:
+            self._remove_bridge(ifname)
+            return result
+        result = self._create_runtime()
         if result != SUCCESS:
             self._remove_bridge(ifname)
         return result
 
     def check_outputs(self):
         # Not implemented yet, just pass the test
+        return SUCCESS
+
+    def run(self):
```

Comment:
this function seems to be mis-named.
You probably want the creation and destruction of the bridge to be in the same function.
Where are the maps created?

Reply:
Maps are created in load_ebpf(). run just launches the runtime. I can move the creation and loading of the environment into the run function